dashboard/pages/primary_screening/callbacks.py

```diff
@@ -84,6 +91,24 @@ def upload_echo_data(contents, names, last_modified, stored_uuid, file_storage):
     )
 
 
+# === STAGE 3 ===
+
+
+def on_stage_3_entry(
```

Comment:
suggestion: rename this callback to use stage name instead of stage index (easier to determine what should happen, and the index itself can be induced from `if current_stage` check)